process/process_darwin.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package process
+
+import (
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/keybase/go-logging"
+	"github.com/keybase/go-updater/command"
+)
+
+// RestartAppDarwin restarts an app. We will still call open if the kill fails.
+func RestartAppDarwin(appPath string, log logging.Logger) error {
+	if appPath == "" {
+		return fmt.Errorf("No app path to restart")
+	}
+	procName := filepath.Join(appPath, "Contents/MacOS/")
+	TerminateAll(procName, log)
```

Comment:
Why is sig term better than calling ctl stop

On Tuesday, May 3, 2016, Gabriel Handford <notifications@github.com> wrote:

> In process/process_darwin.go
> <https://github.com/keybase/go-updater/pull/48#discussion_r61943682>:
>
> > +import (
> > +	"fmt"
> > +	"path/filepath"
> > +	"time"
> > +
> > +	"github.com/keybase/go-logging"
> > +	"github.com/keybase/go-updater/command"
> > +)
> > +
> > +// RestartAppDarwin restarts an app. We will still call open if the kill fails.
> > +func RestartAppDarwin(appPath string, log logging.Logger) error {
> > +	if appPath == "" {
> > +		return fmt.Errorf("No app path to restart")
> > +	}
> > +	procName := filepath.Join(appPath, "Contents/MacOS/")
> > +	TerminateAll(procName, log)
>
> ok talking to steve.. we can wait on doing ctl stop until we implement
> windows, which will come in a separate PR.
> It's not required for darwin/linux since sending SIGTERM is all thats
> needed on those platforms.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/keybase/go-updater/pull/48/files/d9ddece94fc446ecac9e587eb9c1e04df30891f9#r61943682>
>